ethpm_cli/manifest.py

```diff
@@ -99,21 +106,103 @@ def gen_sources(
     )
     if not flag:
         raise Exception("sorry, we dont support specific source selection yet.")
     inline_source_flag = parse_bool_flag(
-        "Would you like to automatically inline all sources? (y/n) "
+        "Would you like to automatically inline all sources?"
     )
     if not inline_source_flag:
         raise Exception("Sorry, we dont support pinning sources yet.")
     # todo: validate a source for every contract type
     return build_sources(contract_types, solc_output, contracts_dir)
 
 
+def gen_deployments(solc_output: Dict[str, Any]) -> Iterable[Callable[..., Manifest]]:
+    flag = parse_bool_flag("Would you like to add a deployment to your package?")
+    if flag:
+        return build_deployments(solc_output)
+    return tuple()
+
+
+def build_deployments(solc_output: Dict[str, Any]) -> Iterable[Callable[..., Manifest]]:
+    deployments_data = gen_all_deployments(solc_output)
+    return (b.deployment(**dep) for dep in deployments_data)
+
+
+@to_tuple
+def gen_all_deployments(solc_output: Dict[str, Any]) -> Iterable[Dict[str, Any]]:
+    while True:
+        yield gen_single_deployment(solc_output)
+        if not parse_bool_flag("Would you like to add another deployment?"):
+            break
+
+
+def gen_single_deployment(solc_output: Dict[str, Any]) -> Dict[str, Any]:
+    chain_id = get_chain_id()
+    w3 = get_w3(chain_id)
+    block_uri = create_latest_block_uri(w3)
+    address = get_deployment_address()
+    contract_type = get_deployment_contract_type(solc_output)
+    # todo: support custom contract instance definitions
+    contract_instance = contract_type
+    tx_hash, block_hash = get_deployment_chain_data(w3)
+    deployment_data = {
+        "block_uri": block_uri,
+        "contract_instance": contract_instance,
+        "contract_type": contract_type,
+        "address": address,
+        "transaction": tx_hash,
+        "block": block_hash,
+    }
+    return {field: value for field, value in deployment_data.items() if value}
+
+
+def get_deployment_chain_data(w3: Web3) -> Tuple[Optional[str], Optional[Any]]:
+    # todo: deployment_bytecode, runtime_bytecode, compiler
+    flag = parse_bool_flag("Do you have a tx hash for your deployment?")
+    if flag:
+        tx_hash = input("Please enter your tx hash. ")
+        tx = w3.eth.getTransaction(tx_hash)
+        return tx_hash, to_hex(tx.blockHash)
+    return (None, None)
+
+
+def get_chain_id() -> int:
+    question = f"On what chain ID is your deployment located? {list(SUPPORTED_CHAIN_IDS.keys())} "
+    chain_id = to_int(text=input(question))
+    while chain_id not in SUPPORTED_CHAIN_IDS.keys():
+        cli_logger.info(f"{chain_id} is not a supported chain id. ")
+        chain_id = to_int(text=input(question))
+    return chain_id
+
+
+def get_deployment_address() -> str:
+    question = "What is the address of your deployment? "
+    address = input(question)
+    while not is_address(address):
```

Comment:
probably want to do `is_checksum_address` here and enforce checksums on user inputs.